### PR TITLE
UAR-331: Adding Update object to model

### PR DESCRIPTION
**Pull request by jkayyala**
Jira [UAR-331](https://companieshouse.atlassian.net/browse/UAR-331)

Adding Update model to node to enable Save and Resume Update functionality

[UAR-331]: https://companieshouse.atlassian.net/browse/UAR-331?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/services/overseas-entities/types.ts [hunk 1]
```diff
@@ -244,6 +246,18 @@ export interface ManagingOfficerCorporate {
     role_and_responsibilities?: string
 }
 
+export interface Update {
+    date_of_creation?: string;
```
Comment (AndreKramer7654) [inline src/services/overseas-entities/types.ts:250]: This needs to be an InputDate

File: src/services/overseas-entities/types.ts [hunk 2]
```diff
@@ -244,6 +246,18 @@ export interface ManagingOfficerCorporate {
...
+    bo_mo_data?: boolean;
```
Comment (AndreKramer7654) [inline src/services/overseas-entities/types.ts:251]: This needs to be a YesNo enum

File: src/services/overseas-entities/mapping.ts [hunk 3]
```diff
@@ -70,7 +73,8 @@ export const mapOverseasEntityResource = (body: OverseasEntityResource): Oversea
             return { ...moi, date_of_birth: mapIsoDate(moi.date_of_birth) }
         }),
         managing_officers_corporate: body.managing_officers_corporate || [],
-        trusts: mapTrustsResource(body.trusts)
+        trusts: mapTrustsResource(body.trusts),
+        update: { ...body.update }
```
Comment (AndreKramer7654) [inline src/services/overseas-entities/mapping.ts:77]: Please define mapper functions.

File: src/services/overseas-entities/types.ts (hunk 2 above)
Comment (AndreKramer7654) [reply src/services/overseas-entities/types.ts:251]: and rename bo_mo_data_fetched

File: src/services/overseas-entities/mapping.ts (hunk 3 above)
Comment (AndreKramer7654) [reply src/services/overseas-entities/mapping.ts:77]: See https://github.com/companieshouse/api-sdk-node/pull/541 - will be adding more fields and they need mapping.

File: test/services/overseas-entities/overseas.entities.spec.ts [hunk 4]
```diff
@@ -314,6 +320,38 @@ describe("Mapping OverseasEntity Tests suite", () => {
...
+            due_diligence: undefined,
+            overseas_entity_due_diligence: undefined,
+            beneficial_owners_statement: undefined,
+            beneficial_owners_individual: undefined,
+            beneficial_owners_corporate: undefined,
+            beneficial_owners_government_or_public_authority: undefined,
+            managing_officers_individual: undefined,
+            managing_officers_corporate: undefined,
+            trusts: undefined,
+            update: mockValues.OVERSEAS_ENTITY_RESOURCE_OBJECT_MOCK.update
```
Comment (AndreKramer7654) [inline test/services/overseas-entities/overseas.entities.spec.ts:337]: could name update resource more specifically

**Review by AndreKramer7654 — APPROVED** [review]

**Review by bantao-ch — APPROVED** [review]

**Review by markpit — APPROVED** [review]

**Pull request merged**